Servo motor library wrapper

tomas321 opened the pull request:
refer to the CHANGELOG for changes.

tomas321 committed:
create Servo library wrapper

tomas321 committed:
update module core to latest version

tomas321 committed:
apply project indent

tomas321 committed:
adjust to API and use stack instead of heap

tomas321 committed:
change uuid type to std string

tomas321 committed:
get use of inheritence and implement poll rating

tomas321 committed:
implement SET_VALUE API datapoint

tomas321 committed:
implement all APIs

tomas321 committed:
add CHANGELOG.md

tomas321 committed:
fix the periodic value update API

tomas321 committed:
add README.md

tomas321 committed:
fix versions and links

tomas321 committed:
finalize defaults and remove redundant logging

stradiot commented:
*  CMakeLists.txt is unnecessary in the repository, it is generated by platformio
*  Comaider actual servo motor icon in wiring.png
*  Servo_motor.hpp move to include/
*  Servo_motor.cpp move to src/
*  Servo_motor.hpp delete epmty line 5
*  Servo_motor.hpp line 15 too long line
*  Servo_motor.hpp #pragma once instead of #ifndef 
*  Servo_motor.hpp Servo_motor.cpp missing #include string
*  Servo_motor.hpp use named parameters
*  Servo_motor.cpp constructor one parameter per line, start on 4
*  When passing std::string pass reference & 
*  Don't move argument objects, instead create new local one
*  this-> is unnecessary in most cases (if there is no conflict with local variables or arguments)
*  left & rightmost position degree can be private static member constants 
*  Methods move_to_rightmost_deg etc can be omitted and more generic write(constant) can be used
*  DRY, most added methods do the same thing and only passing different arguments. One generic method (with parameter) or try catch on top of parent’s write method can be implemented instead (preferred, because it will resolve the same issue with get_current_position too). Is there a case when pin is not attached when you attach it in constructor?
*  member wait is uint32_t, but would fit into uint16_t
*  Int is unnecessary large large return value. Same with resolve_datapoint value parameter. 
*  I like datapoint resolve method inside the class itself. 
*  Servo_motor.hpp constructor ratelimit is the same as poll_rate in cpp?
*  main.cpp 99, 100 unnecessary, just use uuid.c_str() on 101
*  main.cpp 107, 111 indent
*  If resolve_datapoint returns non-zero, are you sure the problem is with datapoint name?
*  Consider changing resolve_datapouint return value type to bool.

tomas321 committed:
incorporate requested changes and refactor code

tomas321 commented:
- Servo_motor.hpp Servo_motor.cpp missing #include string
  - build is successful without it

- `this->` is unnecessary in most cases (if there is no conflict with local variables or arguments)
  - is more comprehensive with it

- left & rightmost position degree can be private static member constants
  - they are in the super class.. this is a override for possible future usage

- Methods move_to_rightmost_deg etc can be omitted and more generic write(constant) can be used
  - if the left and rightmost variables are overridden, than no

- DRY, most added methods do the same thing and only passing different arguments. One generic method (with parameter) or try catch on top of parent’s write method can be implemented instead (preferred, because it will resolve the same issue with get_current_position too). Is there a case when pin is not attached when you attach it in constructor?
  - the pin could happen to be detached manually as failure.. 

- Servo_motor.hpp constructor ratelimit is the same as poll_rate in cpp?
  - yes, i renamed `ratelimit` to `poll_rate`

- main.cpp 107, 111 indent
  - indent and refactoring realted issues are IDE dependent, therefore i will not change them

- If resolve_datapoint returns non-zero, are you sure the problem is with datapoint name?
  - thanks for the heads up, fixed it
